src/LSST/control/PowerSubsystem.cpp

```diff
@@ -473,14 +467,20 @@ SysStatus PowerSubsystem::processDaq(SysInfo const& info, faultmgr::FaultStatusB
         setPowerOff("processDaq had system faults");
     }
 
+    bool reportStateChange = false;
     if (_targPowerStatePrev != _targPowerState || _actualPowerStatePrev != _actualPowerState) {
         LINFO(getClassName(), " power state change prev(targ=", getPowerStateStr(_targPowerStatePrev),
               " act=", getPowerStateStr(_actualPowerStatePrev),
               ") new(targ=", getPowerStateStr(_targPowerState), " act=", getPowerStateStr(_actualPowerState),
               ")");
+        reportStateChange = true;
     }
     _targPowerStatePrev = _targPowerState;
     _actualPowerStatePrev = _actualPowerState;
+    // don't actually report the state change until after the values have been set.
+    if (reportStateChange) {
+        _reportStateChange();
+    }
```

Comment:
If that's the case ("don't actually report the state change until after the values have been set") then consider doing this state transition atomically so that a modified/extended version of the method `_reportStateChange` would also do:
```
void PowerSubsystem::_updateStateChange() {
    _targPowerStatePrev = _targPowerState;
    _actualPowerStatePrev = _actualPowerState;
}

void PowerSubsystem::_updateAndReportStateChange() {
    // First, upgrade the the state variables
    _updateStateChange();
    // Then do report the state
    ....
}
```
Try doing related things together in an atomic way instead of modifying things in a scattered way.

After that, your original algorithm will no longer need `bool reportStateChange` and could be rewritten as:
```
    if (_targPowerStatePrev != _targPowerState || _actualPowerStatePrev != _actualPowerState) {
        ...
        _updateAndReportStateChange();
    } elser {
        _updateStateChange();
    }
   ...
```
An alternative approach if you like it more would be:
```
void _updateStateChange(bool reportStateChange = false) {
    _targPowerStatePrev = _targPowerState;
    _actualPowerStatePrev = _actualPowerState;
    if (reportStateChange) {
        _reportStateChange();
    }
}
```
This would make the following code:
```
    if (_targPowerStatePrev != _targPowerState || _actualPowerStatePrev != _actualPowerState) {
        ...
        bool const reportStateChange = true;
        _updateStateChange(reportStateChange);
    } elser {
        _updateStateChange();
    }
   ...
```

Reply:
I'm not sure what you're after here. Looking at it now, I don't think the `reportStateChange` boolean is fixing anything and can be eliminated.